apps/frontend/src/app/3_organisms/RewardHistory/components/RewardsEarnedHistory/RewardsEarnedHistory.tsx

```diff
@@ -0,0 +1,180 @@
+import React, { FC, useCallback, useEffect, useMemo, useState } from 'react';
+
+import { t } from 'i18next';
+import { nanoid } from 'nanoid';
+
+import {
+  ErrorBadge,
+  ErrorLevel,
+  NotificationType,
+  OrderDirection,
+  OrderOptions,
+  Pagination,
+  Select,
+  Table,
+} from '@sovryn/ui';
+
+import { ExportCSV } from '../../../../2_molecules/ExportCSV/ExportCSV';
+import {
+  DEFAULT_HISTORY_FRAME_PAGE_SIZE,
+  EXPORT_RECORD_LIMIT,
+} from '../../../../../constants/general';
+import { useNotificationContext } from '../../../../../contexts/NotificationContext';
+import { useAccount } from '../../../../../hooks/useAccount';
+import { useMaintenance } from '../../../../../hooks/useMaintenance';
+import { translations } from '../../../../../locales/i18n';
+import { rskClient } from '../../../../../utils/clients';
+import {
+  RewardsEarnedAction,
+  RewardsEarnedHistoryItem_OrderBy,
+  useGetRewardsEarnedHistoryLazyQuery,
+} from '../../../../../utils/graphql/rsk/generated';
+import { dateFormat } from '../../../../../utils/helpers';
+import { decimalic } from '../../../../../utils/math';
+import {
+  RewardHistoryProps,
+  RewardHistoryType,
+} from '../../RewardHistory.types';
+import { rewardHistoryOptions } from '../../RewardHistory.utils';
+import { COLUMNS_CONFIG } from './RewardsEarnedHistory.constants';
+import { generateRowTitle } from './RewardsEarnedHistory.utils';
+import { useGetRewardsEarned } from './hooks/useGetRewardsEarned';
+
+const pageSize = DEFAULT_HISTORY_FRAME_PAGE_SIZE;
+
+export const RewardsEarnedHistory: FC<RewardHistoryProps> = ({
+  selectedHistoryType,
+  onChangeRewardHistory,
+}) => {
+  const { account } = useAccount();
+  const { addNotification } = useNotificationContext();
+  const { checkMaintenance, States } = useMaintenance();
+
+  const [page, setPage] = useState(0);
+
+  const [orderOptions, setOrderOptions] = useState<OrderOptions>({
+    orderBy: 'timestamp',
+    orderDirection: OrderDirection.Desc,
+  });
+
+  const historyAction = useMemo(
+    () =>
+      selectedHistoryType === RewardHistoryType.stakingRevenue
+        ? [RewardsEarnedAction.UserFeeWithdrawn]
+        : [RewardsEarnedAction.StakingRewardWithdrawn],
+    [selectedHistoryType],
+  );
+
+  const { data, loading } = useGetRewardsEarned(
+    account,
+    pageSize,
+    page,
+    orderOptions,
+    historyAction,
+  );
+
+  const [getRewards] = useGetRewardsEarnedHistoryLazyQuery({
+    client: rskClient,
+  });
+
+  const onPageChange = useCallback(
+    (value: number) => {
+      if (data?.length < pageSize && value > page) {
+        return;
+      }
+      setPage(value);
+    },
+    [page, data],
+  );
+
+  const isNextButtonDisabled = useMemo(
+    () => !loading && data?.length < pageSize,
+    [loading, data],
+  );
+
+  const exportData = useCallback(async () => {
```

Comment:
`Staking revenue` CSV export is still not working.

<img width="1960" alt="image" src="https://github.com/DistributedCollective/sovryn-dapp/assets/14878713/ea328493-44f7-4a2f-a55f-08d8827ef24e">